src/transactions/simulation/SimulationTransactionFrame.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/simulation/SimulationTransactionFrame.h"
+#include "ledger/LedgerTxn.h"
+#include "transactions/OperationFrame.h"
+#include "transactions/TransactionUtils.h"
+#include "transactions/simulation/SimulationMergeOpFrame.h"
+
+namespace stellar
+{
+
+TransactionFramePtr
+SimulationTransactionFrame::makeTransactionFromWire(
+    Hash const& networkID, TransactionEnvelope const& envelope,
+    TransactionResult simulationResult)
+{
+    TransactionFramePtr res = std::make_shared<SimulationTransactionFrame>(
+        networkID, envelope, simulationResult);
+    return res;
+}
+
+SimulationTransactionFrame::SimulationTransactionFrame(
+    Hash const& networkID, TransactionEnvelope const& envelope,
+    TransactionResult simulationResult)
+    : TransactionFrame(networkID, envelope), mSimulationResult(simulationResult)
+{
+}
+
+std::shared_ptr<OperationFrame>
+SimulationTransactionFrame::makeOperation(Operation const& op,
+                                          OperationResult& res, size_t index)
+{
+    if (mEnvelope.tx.operations[index].body.type() != ACCOUNT_MERGE)
+    {
+        return OperationFrame::makeHelper(op, res, *this);
+    }
+    else
+    {
+        return std::make_shared<SimulationMergeOpFrame>(
+            op, res, *this, mSimulationResult.result.results()[index]);
+    }
+}
+
+bool
+SimulationTransactionFrame::isTooEarly(LedgerTxnHeader const& header) const
+{
+    return mSimulationResult.result.code() == txTOO_EARLY;
+}
+
+bool
+SimulationTransactionFrame::isTooLate(LedgerTxnHeader const& header) const
+{
+    return mSimulationResult.result.code() == txTOO_LATE;
+}
+
+bool
+SimulationTransactionFrame::isBadSeq(int64_t seqNum) const
+{
+    return mSimulationResult.result.code() == txBAD_SEQ;
```

Comment:
during validation time, you'd never return false here ; only during apply time

Reply:
`SimulationTransactionFrame` isn't used for validation, `applying` is always true on this code path.

Reply:
yes you're right: I thought we would use this simulation code in a "live" environment, but if we do this, we'll need more changes anyways so keeping this PR for "command line replay" is fine for now.